### PR TITLE
Add date_utils for date range localization

### PR DESCRIPTION
**Pull request by Lknechtli**
Add ummalqura calendar support to custom reports date aggregation
Fix gregorian sequential weekly aggregation

### CODE REVIEW

**Comment by Lknechtli** [comment]
Refactoring the crosstabs route out into a separate view - it's up to 800ish lines now.

**Comment by Lknechtli** [comment]
I may have the umalqurra library added as a requirement incorrectly - when re-provisioning, it wasn't added, and I had to do it manually.  Not sure why.

File: app/data/localization/date_utils.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+            end_month = month + 1
+            if end_month >= 12:
+                end_year = year+1
+                end_month = 1
+
+            end_date_um = HijriDate(end_year, end_month, 1)
+            end_year_gr = int(end_date_um.year_gr)
+            end_month_gr = int(end_date_um.month_gr)
+            end_day_gr = int(end_date_um.day_gr)
+            end_date_gr = datetime(end_year_gr, end_month_gr, end_day_gr)
```
Comment (ddohler) [inline app/data/localization/date_utils.py]: Could this be generated using the `hijri_to_gregorian` function above?

Comment (Lknechtli) [inline app/data/localization/date_utils.py]: yup

**Comment by kshepard** [comment]
@Lknechtli: I also had an umalqurra import issue after provisioning, but restarting the driver-app service fixed it. So it doesn't look like an issue with how the requirement is added.

File: app/data/views.py [hunk 2]
```diff
@@ -649,6 +668,160 @@ def _make_period_case(self, period_type, request, queryset):
...
+            'month_of_year': {
+                'type': 'derived',
+                'query': hijri_month_range,
+            },
+            'week_of_year': {
+                'type': 'derived',
+                'query': hijri_week_range
+            },
+            'day_of_week': {
+                'type': 'source',
```
Comment (ddohler) [inline app/data/views.py]: I agree that these could be better named; perhaps `builtin` and `generated`?

**Comment by kshepard** [comment]
It seems the 'day' aggregation in gregorian doesn't work (returns all zeros), yet the ummalqura version is fine. Think you'd be able to take a look at that while your head is still deep in this code?

**Comment by ddohler** [comment]
+1 after squashing commits; it's a bit tough to tell if everything is working perfectly, but the Umm al-Qura aggregations seem to give roughly the same order of magnitude of events as their Gregorian counterparts.

**Comment by kshepard** [comment]
+1 from me too. Code looks good and numbers seem to be in the right ballpark. Nice work!

**Comment by Lknechtli** [comment]
closes #569

**Pull request merged**